python/tk_desktop/desktop_window.py

```diff
@@ -40,6 +42,7 @@
 from .project_commands_model import ProjectCommandModel
 from .project_commands_model import ProjectCommandProxyModel
 from .project_commands_widget import ProjectCommandDelegate
+from shotgun_desktop.desktop_login import get_shotgun_authenticator
```

Comment:
Maybe this waits for the upgradable bootstrap logic for desktop internal, but shouldn't this be pulling from something like the login framework so that we can centralize the code that does the caching of these values?  That way in the future as we have better support for multiple logins and encrypted storage it will be easier to update?

Seems like ideally no authentication logic would be in tk-desktop or tk-desktop-internal.  Is there a way we can implement this to get closer to that?